src/services/models/ajaxModels.js

```diff
@@ -11,10 +12,34 @@ export class AjaxError {
   constructor(ajaxError) {
     this.ajaxError = ajaxError;
     const resp = ajaxError.response || {};
-    this.errorMessage = resp.ExceptionMessage || resp.Message || resp.Error || ajaxError.message || `An unknown error occurred`;
+
+    // In Java, the ExceptionMessage contains a generic message and info about the request.
+    // But the InnerExceptionMessage message looks to contain information about the actual error.
+    this.errorMessage = resp.InnerExceptionMessage || resp.ExceptionMessage || resp.Message || resp.Error || ajaxError.message || `An unknown error occurred`;
     this.status = ajaxError.status;
+
     // Log all errors in the console
     console.error(ajaxError);
+
+    // For general application errors, figure out a more detailed message if possible.
+    // In dotNet, the ExceptionMessage may contain JSON that can be further parsed.
+    if (this.status >= 500 && this.errorMessage) {
+      this[parseApplicationMessage](((JSON.parse(this.errorMessage) || {}).Message) || this.errorMessage);
```

Comment:
I don't see the value of using the symbol here. Just seems to make things more confusing.

Reply:
That seemed like the way to make it private... but if we aren't going to translate... then its not going to be needed anyway.